src/snowflake/snowpark/mock/_connection.py

```diff
@@ -83,6 +84,18 @@ def _build_target_path(stage_location: str, dest_prefix: str = "") -> str:
     return f"{qualified_stage_name}{dest_prefix_name if dest_prefix_name else ''}"
 
 
+def _row_output_converter(cell_value, column_idx, sf_types):
```

Comment:
Why do we need the conversion here? The data of a ColumnEmulator should be coerced into the column type when  an instance of ColumnEmulator is initialized.

Reply:
Adam and I chatted about this offline, we need this conversion logic because when there are multiple json files with conflicting schema, for example, the same column name maps to an number in one file and an array in another file. When Snowflake sees a conflict like this, the default behavior seems to be just converting all values in this column to string.

Reply:
moved the logic into an inner method: convert_value_to_str_for_string_type